sound/soc/sof/trace.c

```diff
@@ -17,20 +17,29 @@ static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
 				   loff_t pos, size_t buffer_size)
 {
 	wait_queue_entry_t wait;
+	unsigned long flags;
+	size_t ret;
+
+	spin_lock_irqsave(&sdev->trace_lock, flags);
```

Comment:
This function cannot (or at least must not) be called from an atomic context, so, there's no need to save IRQ flags, IRQs aren't disabled here. Therefore spin_(un)lock_irq() would suffice.